internal/services/inference/deployment.go

```diff
@@ -0,0 +1,349 @@
+		DeleteContext: ResourceDeploymentDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create:  schema.DefaultTimeout(defaultInferenceDeploymentTimeout),
+			Read:    schema.DefaultTimeout(defaultInferenceDeploymentTimeout),
+			Update:  schema.DefaultTimeout(defaultInferenceDeploymentTimeout),
+			Delete:  schema.DefaultTimeout(defaultInferenceDeploymentTimeout),
+			Default: schema.DefaultTimeout(defaultInferenceDeploymentTimeout),
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Optional:    true,
+				Description: "The deployment name",
+			},
+			"region":     regional.Schema(),
+			"project_id": account.ProjectIDSchema(),
+			"node_type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The node type to use for the deployment",
+			},
+			"model_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The model name to use for the deployment",
+			},
+			"model_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The model id used for the deployment",
+			},
+			"accept_eula": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Whether or not the deployment is accepting eula",
+			},
+			"tags": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "The tags associated with the deployment",
+			},
+			"min_size": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: "The minimum size of the pool",
+			},
+			"max_size": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: "The maximum size of the pool",
+			},
+			"size": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "The size of the pool",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the deployment",
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the creation of the deployment",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the last update of the deployment",
+			},
+
+			"private_endpoint": {
+				Type:         schema.TypeList,
+				Optional:     true,
+				MaxItems:     1,
+				AtLeastOneOf: []string{"public_endpoint"},
+				Description:  "List of endpoints",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeString,
+							Description: "The id of the private endpoint",
+							Computed:    true,
+						},
+						"private_endpoint_id": {
```

Comment:
```suggestion
						"private_network_id": {
```
You will also need to change in the rest of the code.